data_tools/compute_coding_progress.py

```diff
@@ -5,10 +5,13 @@
 from time import gmtime, strftime
 
 
-def compute_coding_progress(dataset_id, force_recount=False):
+def compute_shard_coding_progress(dataset_id, shard_index=None, force_recount=False):
     """Compute and return the progress metrics for a given dataset.
     This method will initialise the counts in Firestore if they do
     not already exist."""
+    if shard_index is not None and shard_index != 1:
+        dataset_id += f'_{shard_index}'
```

Comment:
I'm not really convinced by this naming scheme of having the first shard not have a number. I kinda get why you would do this, but I think it would be better to have either 'sharded' or 'unsharded' datasets. For unsharded ones, there is no number, for sharded one every shard has a number

Reply:
I thought that Firestore didn't support document renaming, so this seemed less risky than copying and deleting a dataset which someone might already be working on when we fill up the first shard.

Unless you'd like all the datasets to be littered with "_1"s, even the smaller ones which make up the vast majority of datasets at the moment, and to make backwards-compatibility with existing datasets much more painful.

Reply:
Right - this makes more sense given comment below that they're not actually shards, they're more continuations